boot/zephyr/Kconfig

```diff
@@ -248,6 +248,18 @@ config BOOT_SERIAL_DETECT_PIN_VAL
 	  Logic value of the detect pin which triggers serial recovery
 	  mode.
 
+# Workaround for not being able to have commas in macro arguments
+DT_CHOSEN_Z_CONSOLE := zephyr,console
+
+config RECOVERY_UART_DEV_NAME
+	string "UART Device Name for Recovery UART"
+	default "$(dt_chosen_label,$(DT_CHOSEN_Z_CONSOLE))" if HAS_DTS
```

Comment:
Is there any reason not to use DT_CHOSEN_Z_CONSOLE directly? Are there any platforms we're interested in here that have CONFIG_HAS_DTS=n? If so, I think this patch seems like making the best of a bad situation, but otherwise I don't understand why we have to have this kconfig option.

Reply:
I might `CONFIG_HAS_DTS=n` PLATFORM, that reasonable.
Yeah I introduce something which allow to get the same DTS configuration to mcuboot. Not possible without kconfig i believe. Do you think I should just declare stand-alone config property here, UART_) by default?

Reply:
> Is there any reason not to use DT_CHOSEN_Z_CONSOLE directly? Are there any platforms we're interested in here that have CONFIG_HAS_DTS=n? If so, I think this patch seems like making the best of a bad situation, but otherwise I don't understand why we have to have this kconfig option.

not sure I follow, what do you mean by use it directly?

Reply:
@galak I believe directly mean "get string constructed by syntax: `"$(dt_chosen_label,$(DT_CHOSEN_Z_CONSOLE))"` above into C code, without kconfig as the intermediary.